### DISCUSSION TITLE
Output encoding problem?

### DISCUSSION THREAD

**Discussion by tomsnunes** [ggml-org/llama.cpp#366] [Q&A] [answered]
Hey guys! When I run the binary compiled using code from the master branch on Windows 11, during interaction with the model, in situations involving Latin characters such as á, ã, é as a response, the characters are displayed incorrectly.
I got the same kind of behavior when trying to interact with the model in languages like Portuguese and Spanish. Characters from languages such as Japanese, Chinese, and Korean are also displayed incorrectly.
I believe it's not a problem with my terminal's encode configuration, as the characters are displayed correctly when inputting data to the model, but the output from model brings invalid characters.

My system information:
CPU: Ryzen 9 5900x
OS: Windows 11 Pro x64
cmake version 3.26.0
Python 3.10.10
Compiler: Visual Studio Community 2022 using compilers for 17.5.2

Example:
![image](https://user-images.githubusercontent.com/8005536/226700621-ebc50e53-e514-48a0-97b5-cf74a0791ee1.png)

Is this behaviour expected?
Thanks!

**Comment by j-f1** [comment, marked as answer]
That output suggests to me that your terminal is not using the UTF-8 encoding. Make sure it’s configured to do that — LLaMA was trained to output UTF-8 text.

**Reply by tomsnunes** [reply]
Looks like you're right. After some further research, I had to turn my regional settings on to use the beta feature that allows the system locale (language for non-Unicode programs) to UTF-8 like that:

![image](https://user-images.githubusercontent.com/8005536/226707023-f5419829-ae12-46e5-a6e7-1b8c70f1d65c.png)

Thanks for the help!
